pkg/apiserver/options.go

```diff
@@ -0,0 +1,66 @@
+package apiserver
+
+import (
+	compressGzip "compress/gzip"
+	ginGzip "github.com/gin-contrib/gzip"
+	"github.com/gin-gonic/gin"
+	"io"
+)
+
+type optionState struct {
+	router *gin.Engine
+}
+
+type Option func(*optionState)
+
+type CompressionLevel int
+
+const (
+	BestCompression    CompressionLevel = ginGzip.BestCompression
+	BestSpeed          CompressionLevel = ginGzip.BestSpeed
+	DefaultCompression CompressionLevel = ginGzip.DefaultCompression
+	NoCompression      CompressionLevel = ginGzip.NoCompression
+)
+
+func WithCompression(level CompressionLevel, gzipOpts ...ginGzip.Option) Option {
```

Comment:
Do we want to have this as an option we set via code or as a config setting?  
As we have all other settings like port etc from the config, wouldn't it more consistent to configure this via config, too?  
Or am I missing something here?

Reply:
It would be more consistent. I just have to transform all the config options then from the config to `ginGzip.Option`s, but yeah, that would be much better. And that shouldn't be too hard... Will do that tomorrow

Reply:
Actually did this now